### PR TITLE
requester_pays feature

### PR DESCRIPTION
**Pull request by nturaga**
Example: 

Considering you have ` gcloud config list project` set to a valid billing project, then,

```
gsutil_ls("gs://bioconductor-full-devel", requester_pays = TRUE)

install(c("BiocParallel", "GenomicRanges"), requester_pays=TRUE)
```

The user gets billed instead of the host that is storing the data.

If you do not have a billing project set,

```
gcloud config set project <project_id>
```

### CODE REVIEW

File: R/gsutil.R [hunk 1]
```diff
@@ -102,16 +102,36 @@ NULL
...
 }
 
+## Get gsutil billing project
+.gsutil_billing_project <-
+    function()
+{
+    ## Get gcloud binary
+    gcloud <- .gsutil_find_binary('gcloud')
+    args <- c('config', 'list', 'project')
+    res <- system2(gcloud, args, stdout = TRUE, stderr=TRUE)[2]
```
Comment (mtmorgan) [inline R/gsutil.R]: More robust to ask for `--format=json` and use `jsonlite::fromJSON()$core$project` ?

Need to check for warnings / errors.

File: R/gsutil.R [hunk 2]
```diff
@@ -102,16 +102,36 @@ NULL
...
+    function(args, requester_pays = FALSE)
 {
     gsutil <- .gsutil_find_binary("gsutil")
     stopifnot(file.exists(gsutil))      # bad environment variables
 
     wmsg <- NULL
+    if (requester_pays)
+        args <- c("-u", .gsutil_billing_project(), args)
     value <- withCallingHandlers(tryCatch({
-        system2(gsutil, args, stdout = TRUE, stderr = TRUE, wait=TRUE)
```
Comment (mtmorgan) [inline R/gsutil.R]: please revert whitespace only changes

File: R/gsutil.R [hunk 3]
```diff
@@ -207,7 +232,7 @@ gsutil_exists <-
     gsutil <- .gsutil_find_binary("gsutil")
     stopifnot(file.exists(gsutil))      # bad environment variables
 
-    vapply(source, .gsutil_exists_1, logical(1), gsutil)
+    vapply(source, .gsutil_exists_1, logical(1), gsutil, requester_pays=requester_pays)
```
Comment (mtmorgan) [inline R/gsutil.R]: avoid long lines here and elsewhere; format as
```
    vapply(
        source, .gsutil_exists_1, logical(1), gsutil,
        requester_pays=requester_pays
   )

**Comment by nturaga** [comment]
Thank you @mtmorgan I'll make the changes.

**Comment by nturaga** [comment]
Closing this issue as the feature `requester_pay` needs to go on a new branch

**Pull request closed without merging**